vllm/model_executor/layers/fused_moe/fused_moe.py

```diff
@@ -1564,3 +1565,70 @@ def fused_moe(
                          a1_scale=a1_scale,
                          a2_scale=a2_scale,
                          block_shape=block_shape)
+
+
+def cutlass_moe(
+    a_q: torch.Tensor,
+    a_scale: torch.Tensor,
+    w1_q: torch.Tensor,
+    w2_q: torch.Tensor,
+    w1_scale: torch.Tensor,
+    w2_scale: torch.Tensor,
+    topk_weights: torch.Tensor,
+    topk_ids: torch.Tensor,
+    m: int,
+    n: int,
+    k: int,
+    num_groups: int,
+    ab_strides1: torch.Tensor,
+    c_strides1: torch.Tensor,
+    ab_strides2: torch.Tensor,
+    c_strides2: torch.Tensor,
+    intermediate_scale: Optional[torch.Tensor] = None,
+):
```

Comment:
A few nits: 
Could you please add a doc comment regarding the expectation of shapes and datatypes for weights, scales and hidden states? Eg: We could also have block scale flavor for cutlass based FP8 MoE, so it would be nice to clarify. Preferably call it `cutlass_fp8_moe` or a version of that. Also consistently use num_experts/num_groups.

Reply:
+1